nb2/bot/slack_bot.py

```diff
@@ -148,44 +168,61 @@ def help(self):
-            ">`nb2 remind (me|<person>) of <person>` Digs up a memorable quote from the past, and "
-            "remind the person."
+            ">`nb2 remind (me|<@person>) of <@person>` Digs up a memorable quote from the past, "
+            "and remind the person."
         )
 
         blocks = [{"type": "section", "text": {"type": "mrkdwn", "text": msg}}]
 
         return self.Result(ok=True, message=blocks)
 
-    def remember(self, target_slack_user_id: str, quote: str):
+    def remember(self, message: str) -> Result:
         """
-        Add quote to a NB's memory of a Person with target_slack_user_id, if they exist.
+        Add quote to a NB's memory of a Person, if they exist.
         If they don't exist, look up their info and create a new Person in memory first.
 
         Args:
-            target_slack_user_id: string representing slack user id for Person to insert
-                                  quote into.
-            quote: string representing content of Quote to store.
+            message: A message that was sent to the bot.
 
         Returns:
             A Result namedtuple.
         """
-        if not get_person_by_slack_user_id(target_slack_user_id):
-            user_info = self.fetch_user_info(target_slack_user_id)
-            name = user_info.get("name")
-            first_name = name.split()[0]
+        matched = re.match(self.REMEMBER_REGEX, message, re.I)
 
-            create_person_dto = CreatePersonDTO(
-                slack_user_id=target_slack_user_id, first_name=first_name
-            )
+        target_user_id = trim_mention(matched.group("user_id"))
+        quote = matched.group("quote")
+        person, is_active = get_person(target_user_id)
 
-            create_person(create_person_dto)
-
-        add_quote_dto = AddQuoteDTO(slack_user_id=target_slack_user_id, content=quote)
+        # The Person doesn't exist, so create a new Person from information fetched
+        # from Slack. If the user id could not be found in Slack, create a ghost Person.
+        if person is None:
+            try:
+                user_info = self.fetch_user_info(target_user_id)
+                user_profile = user_info["profile"]
+                create_person_dto = CreatePersonDTO(
+                    slack_user_id=target_user_id,
+                    first_name=user_profile.get("first_name"),
+                    last_name=user_profile.get("last_name"),
+                    ghost_user_id=user_profile.get("display_name") or user_info.get("name"),
+                )
+            except SlackApiError:
+                create_person_dto = CreateGhostPersonDTO(
+                    # TODO: Need a better way to manage first_name in this case
+                    ghost_user_id=target_user_id,
+                    first_name=target_user_id,
+                )
+
+            person = create_person(create_person_dto)
+        else:
+            if is_active:
+                person = self.update_display_name(person)
```

Comment:
this will only update the ghost_id and not the first/last name. Do we not want to include that?
Also we'll need to confirm that you can't update the ghost_id to something that's already in use

Reply:
True, I was thinking people don't often update their names but that's still totally a legit and possible use case, will add